test/spec/features/stack/install_spec.rb

```diff
@@ -142,5 +142,14 @@
       k = run 'kontena stack ls -q'
       expect(k.out.split(/[\r\n]/)).to match array_including('twemproxy', 'twemproxy-redis_from_registry', 'twemproxy-redis_from_yml')
     end
+
+    it 'does not mutate the $STACK variable' do
+      with_fixture_dir("stack/depends") do
+        k = run 'kontena stack install'
+        expect(k.code).to eq (0)
+      end
+      k = run 'kontena service show twemproxy/twemproxy'
+      expect(k.out).to match(/STACKNAME=twemproxy$/m)
```

Comment:
Something odd with the  `/...$/m` regexp?

```
  1) stack install For a stack with dependencies does not mutate the $STACK variable
     Failure/Error: expect(k.out).to match(/STACKNAME=twemproxy$/m)
     
       expected "e2e/twemproxy/twemproxy:\r\n  created: 2017-12-18T14:23:19.371Z\r\n  updated: 2017-12-18T14:23:19.66...r\n          ip: 10.81.128.66\r\n          public ip: 82.181.8.229\r\n          status: running\r\n" to match /STACKNAME=twemproxy$/m
       Diff:
       @@ -1,2 +1,34 @@
       -/STACKNAME=twemproxy$/m
       +e2e/twemproxy/twemproxy:
       +  created: 2017-12-18T14:23:19.371Z
       +  updated: 2017-12-18T14:23:19.666Z
       +  stack: e2e/twemproxy
       +  desired_state: running
       +  image: ianblenke/twemproxy:latest
       +  revision: 1
       +  stack_revision: 1
       +  stateful: no
       +  scaling: 1
       +  strategy: ha
       +  read_only: no
       +  stop_grace_period: 10s
       +  deploy_opts:
       +  dns: twemproxy.twemproxy.e2e.kontena.local
       +  env: 
       +    - REDIS1_PORT_6379_TCP_ADDR=twemproxy-redis_from_registry
       +    - REDIS2_PORT_6379_TCP_ADDR=twemproxy-redis_from_yml
       +    - STACKNAME=twemproxy
       +  ports:
       +    - 6000:6000/tcp
       +  instances:
       +    twemproxy/twemproxy/1:
       +      scheduled_to: localhost
       +      deploy_rev: 2017-12-18 14:23:20 UTC
       +      rev: 2017-12-18 14:23:20 UTC
       +      state: running
       +      containers:
       +        twemproxy.twemproxy-1 (on localhost):
       +          dns: twemproxy-1.twemproxy.e2e.kontena.local
       +          ip: 10.81.128.66
       +          public ip: 82.181.8.229
       +          status: running
       
     # ./spec/features/stack/install_spec.rb:152:in `block (3 levels) in <top (required)>'
```

Reply:
`expect(k.out.lines.map(&:strip)).to include "STACKNAME=twemproxy"`? :)

Reply:
The `/.../m` flag doesn't seem to to matter, but the `\r` in the output seems to make the `$` not match:

```
irb(main):014:0> "   - FOO=bar\r\n".match /FOO=bar$/
=> nil
irb(main):015:0> "   - FOO=bar\n".match /FOO=bar$/
=> #<MatchData "FOO=bar">
```